enterprise_access/apps/api/v1/views/subsidy_access_policy.py

```diff
@@ -785,25 +792,26 @@ def can_redeem(self, request, enterprise_customer_uuid):
             if redeemable_policies:
                 resolved_policy = SubsidyAccessPolicy.resolve_policy(redeemable_policies)
 
-            try:
-                if resolved_policy:
-                    list_price = resolved_policy.get_list_price(lms_user_id, content_key)
-                elif successful_redemptions:
-                    # Get the policy record used at time of successful redemption.
-                    # [2023-12-05] TODO: consider cleaning this up.
-                    # This is kind of silly, b/c we only need this policy to compute the
-                    # list price, and it's really only *necessary* to fetch that price
-                    # from within the context of a *policy record* for cases where that successful
-                    # policy was assignment-based (because the list price for assignments might
-                    # slightly different from the current list price in the canonical content metadata).
-                    successfully_redeemed_policy = self.get_queryset().filter(
-                        uuid=successful_redemptions[0]['subsidy_access_policy_uuid'],
-                    ).first()
-                    list_price = successfully_redeemed_policy.get_list_price(lms_user_id, content_key)
-            except ContentPriceNullException as exc:
-                raise RedemptionRequestException(
-                    detail=f'Could not determine list price for content_key: {content_key}',
-                ) from exc
+            if resolved_policy:
+                list_price = resolved_policy.get_list_price(lms_user_id, content_key)
+            elif successful_redemptions:
+                # Get the policy used for redemption and use that to compute the price. If the redemption was the result
+                # of assignment, the historical assignment price might differ from the canonical price. We prefer to
+                # display the redeemed price to avoid confusion.
+                successfully_redeemed_policy = policy_by_redemption_uuid[successful_redemptions[0]['uuid']]
+                list_price = successfully_redeemed_policy.get_list_price(lms_user_id, content_key)
+            else:
```

Comment:
Here's the red "else" in the corresponding diagram:

![Screenshot 2025-02-26 at 4 50 18 PM](https://github.com/user-attachments/assets/4ead368e-41c1-4292-9916-69f9ca825a95)